util/grub.d/00_menu_auto_hide.in

```diff
@@ -1,15 +1,20 @@
 #! /bin/sh
 
-# Disable / skip generating menu-auto-hide config parts on serial terminals
-for x in ${GRUB_TERMINAL_INPUT} ${GRUB_TERMINAL_OUTPUT}; do
-  case "$x" in
-    serial*)
-      exit 0
-      ;;
-  esac
-done
-
 cat << EOF
+# Boot Counting
+if [ "\${boot_counter}" ]; then
+  if [ "\${boot_success}" = "0"]; then
```

Comment:
Minor: can we just collapse these two as:

```
if [ "\${boot_counter}" -a "\${boot_success}" = "0" ]; then
```

?

Reply:
Hmm, I'm not sure I understand this. Isn't

```
if A then
  if B then
    ...
  fi
fi
```

logically equivalent to:

```
if A and B then
  ...
fi
```

?

Reply:
@jlebon yes it is, but you are missing the else part in your example:

```
if A then
    if B then
    ...
    else
    ...
    fi
fi
```

```
if A and B then
    ...
else
    ...
fi
```

The else statement will be executed even `if not A` so then you need:

```
if A and B then
    ...
elif not A then
    ...
fi
```

Reply:
There is no `else` on the first two if-statements though, only the third. :)
Anyway, it's just a minor nit; not really worth discussing it more than we already have.

Reply:
@jlebon sorry, I misread and thought you were talking about the second if-statement. Yes, for the first one you are correct! cc @LorbusChris